dreamerv2/common/replay.py

```diff
@@ -297,33 +296,34 @@ def _sample_sequence(self, oversampling):
             e_unc = np.exp(uncertainties - np.max(uncertainties))
             uncertainty_norm = e_unc / e_unc.sum()
             episode_key = np.random.choice(
-                list(self._episodes_uncertainties.keys()),
+                uncertainties,
                 p=uncertainty_norm
             )
             self._logger.scalar("replay/uncertainty", self._episodes_uncertainties[episode_key])
         elif oversampling:
-            episodes_0 = []
-            episodes_1 = []
-
-            for epi in episodes:
-                if epi["task_index"] == 0:
-                    episodes_0.append(epi)
-                elif epi["task_index"] == 1:
-                    episodes_1.append(epi)
-
-            if np.random.uniform(0, 1) < 0.99 and len(episodes_1) > 0:
-                i = np.random.randint(0, len(episodes_1))
-                episode = episodes_1[i]
-            elif len(episodes_0) > 0:
-                i = np.random.randint(0, len(episodes_0))
-                episode = episodes_0[i]
+            indexes = list(self._task_index.values())
+            # if there is a mismatch in lengths lets sync the rewards with self._complete_eps()
+            if len(indexes) != len(episodes_keys):
+                _, _, _, self._task_index = load_episodes(self._directory,
+                                                          self._capacity, self._minlen, self._coverage_sampling,
+                                                          self._coverage_sampling_args, check=False)
+                indexes = list(self._task_index.values())
+            zeros = indexes.count(0)
```

Comment:
This is perfectly okay, but we can also use [numpy.unique](https://numpy.org/doc/stable/reference/generated/numpy.unique.html), so you don't have to count each possible value separately.

Reply:
With `return_counts=True` obviously

Reply:
Ok, i will check it out, thanks.